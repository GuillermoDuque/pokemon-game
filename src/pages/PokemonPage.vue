<template>
<h1 v-if="!pokemon">Espere por favor...</h1>
<div v-else>
 <h1>¿Quién es ese pokemon?</h1>
  <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon" />
  <PokemonOption :pokemons="pokemonArr"/>
</div>
 
</template>
<script>
import PokemonPicture from '@/components/PokemonPicture.vue'
import PokemonOption from '@/components/PokemonOption.vue'

import getPokemonOptions from '@/helpers/getPokemonOptions'

console.log( getPokemonOptions() )

export default {
    components:{PokemonPicture, PokemonOption},
    data(){
      return{
        pokemonArr: [],
        pokemon: null, 
        showPokemon: false
      }
    },
    methods:{
      async mixPokemonArray(){
          this.pokemonArr = await getPokemonOptions() 
          
          const rndInt = Math.floor( Math.random() * 4)
          this.pokemon = this.pokemonArr[rndInt]
      }
    },
    mounted(){
      this.mixPokemonArray()
    }

}
</script>

