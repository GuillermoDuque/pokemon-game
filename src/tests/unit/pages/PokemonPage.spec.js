import {shallowMount} from '@vue/test-utils'
import PokemonPage  from '@/pages/PokemonPage'
import { mockPokemons } from '../mocks/pokemons.mock'
 /* eslint-disable no-undef */
describe('PokemonPageComponent', () =>{

    let wrapper

    beforeEach(() => {
        wrapper = shallowMount( PokemonPage )
    }) 
    
   

    test('debe de hacer match con el snapshot', () =>{
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('debe de llamar al mixPokemonArray al montar', () =>{
        const mixPokemonArraySpy = jest.spyOn( PokemonPage.methods, 'mixPokemonArray' )
        shallowMount(PokemonPage)
        expect( mixPokemonArraySpy ).toHaveBeenCalled
    })

    test('debe de hacer match con el snapshot cuando cargan los pokemons', () =>{
        const wrapper = shallowMount(PokemonPage, {
            data(){
                return{
                    pokemonArr: mockPokemons,
                    pokemon: mockPokemons[0], 
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })
  
    test('debe de mostrar los componentes de pokemonPicture y pokemonOption', () =>{
        const wrapper = shallowMount(PokemonPage, {
            data(){
                return{
                    pokemonArr: mockPokemons,
                    pokemon: mockPokemons[0], 
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        const picture = wrapper.find('pokemon-picture-stub')
        const options= wrapper.find('pokemon-option-stub')
         
        expect(picture.exists()).toBeTruthy
        expect(options.exists()).toBeTruthy

        expect(picture.attributes('pokemonid')).toBe('1')
        expect(options.attributes('pokemons')).toBeTruthy()

    })


})