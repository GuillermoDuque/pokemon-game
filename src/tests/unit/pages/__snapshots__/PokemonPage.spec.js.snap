// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPageComponent debe de hacer match con el snapshot 1`] = `<h1>Espere por favor...</h1>`;

exports[`PokemonPageComponent debe de hacer match con el snapshot cuando cargan los pokemons 1`] = `
<div>
  <h1>¿Quién es ese pokemon?</h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-option-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-option-stub>
  <!--v-if-->
</div>
`;
